### PR TITLE
PRC-468 Add enter phone numbers page to Add address journey

### PR DESCRIPTION
**Pull request by bill-moj**
PRC-468

image

image

### CODE REVIEW

File: server/routes/contacts/manage/addresses/add-address-phone/AddAddressPhonesSchema.ts [hunk 1]
```diff
@@ -20,3 +20,35 @@ export const phonesSchema = async (req: Request<unknown, unknown, { save?: strin
...
+      save?: string
+      phones?: {
+        type?: string
+        phoneNumber?: string
+        extension?: string
+      }[]
+    }
+  >,
+) =>
+  createSchema({
```
Comment (james-baker-moj) [inline server/routes/contacts/manage/addresses/add-address-phone/AddAddressPhonesSchema.ts:38]: Can you add a jest test that checks the empty forms are not added to the resulting array please?
That keeps our controller nice and safe.

File: server/routes/contacts/manage/addresses/add-address-phone/AddAddressPhonesSchema.ts [hunk 2]
```diff
@@ -49,6 +51,18 @@ export const optionalPhonesSchema = async (
     save: z.string().optional(),
     add: z.string().optional(),
     remove: z.string().optional(),
-  })
+  }).transform(({ phones, save, add, remove }) => ({
+    phones: isAllBlank
```
Comment (james-baker-moj) [inline server/routes/contacts/manage/addresses/add-address-phone/AddAddressPhonesSchema.ts:64]: nice

**Review by james-baker-moj — APPROVED** [review]
LGTM
